src/core/a-entity.js

```diff
@@ -461,22 +467,23 @@ var proto = Object.create(ANode.prototype, {
       var component = this.components[attr] || components[attr];
       var partialComponentData;
       value = value === undefined ? '' : value;
-      var valueStr = value;
+      var deserializedValue = value;
       var oldValue;
+      var serializedValue = value;
 
       if (component) {
         if (typeof value === 'string' && componentPropValue !== undefined) {
           // Update currently-defined component data with the new property value.
           partialComponentData = self.getAttribute(attr) || {};
           partialComponentData[value] = componentPropValue;
-          value = partialComponentData;
+          deserializedValue = partialComponentData;
```

Comment:
**s**erialized a **s**tring, deserialized is an obj. i think it's a common term in APIs

 i changed it because `valueStr` told me it was a string, but knowing the type wasn't telling me what it was supposed to be used for or what it represented.